services/openshiftmisc/src/handlers/routeMigration.js

```diff
@@ -0,0 +1,275 @@
+  });
+  const openshift = new OpenShiftClient.OApi({
+    url: openshiftConsole,
+    insecureSkipTlsVerify: true,
+    auth: {
+      bearer: openshiftToken
+    }
+  });
+
+  // check that the namespaces exist for source and destination before we try and move any routes
+  try {
+    // check source
+    const projectSourceGet = promisify(openshift.projects(openshiftProject).get, { context: openshift.projects(openshiftProject) })
+    projectStatus = await projectSourceGet()
+    logger.info(`${openshiftProject}: Project ${openshiftProject} already exists, continuing`)
+    // check dest
+    const projectDestGet = promisify(openshift.projects(destinationOpenshiftProject).get, { context: openshift.projects(destinationOpenshiftProject) })
+    projectStatus = await projectDestGet()
+    logger.info(`${openshiftProject}: Project ${destinationOpenshiftProject} already exists, continuing`)
+  } catch (err) {
+    // throw error if the namespace doesn't exist
+    logger.error(err)
+    throw new Error
+  }
+
+  // @TODO: this seems a bit silly, might be a better way to do it. but `.patch` on the routemigrates resource fails with,
+  // research says this is because crd is not supported to be patched
+  // `message=the body of the request was in an unknown format - accepted media types include: application/json-patch+json`
+  try {
+    const migrateRoutesDelete = promisify(
+      dioscuri.ns(openshiftProject).routemigrates(openshiftProject).delete
+    );
+    await migrateRoutesDelete({
+      body: {}
+    });
+    await new Promise(resolve => setTimeout(resolve, 10000)); // sleep a bit after deleting
+    try {
+      const migrateRoutesPost = promisify(
+          dioscuri.ns(openshiftProject).routemigrates.post
+      );
+      await migrateRoutesPost({
+        body: migrateRoutes(openshiftProject, destinationOpenshiftProject)
+      });
+      logger.verbose(`${openshiftProject}: created routeMigration resource`);
+    } catch (err) {
+        logger.error(err);
+        throw new Error();
+    }
+  } catch (err) {
+    await new Promise(resolve => setTimeout(resolve, 1000)); // sleep a bit before creating
+    try {
+      const migrateRoutesPost = promisify(
+          dioscuri.ns(openshiftProject).routemigrates.post
+      );
+      await migrateRoutesPost({
+        body: migrateRoutes(openshiftProject, destinationOpenshiftProject)
+      });
+      logger.verbose(`${openshiftProject}: created routeMigration resource`);
+    } catch (err) {
+        logger.error(err);
+        throw new Error();
+    }
+  }
+
+
+  sendToLagoonLogs(
+    'info',
+    projectName,
+    '',
+    'task:misc-openshift:route:migrate',
+    data,
+    `*[${projectName}]* Route Migration between environments *${destinationOpenshiftProject}* started`
+  );
+  // check the route migrate resource for the status conditions, only update lagoon on a completed task
+  var whileCount = 0;
+  var breakLoop = false;
+  try {
+    while (whileCount < 10 && !breakLoop) {
```

Comment:
This while loop will totally block the node event loop, so the other tasks that this service does will have to wait for the route migration to finish.

Since we are just updating the API at this point, we could move all this code into it's own function, and call it over and over again in a `setTimeout` as needed until it's finished.

Reply:
Sounds good, is there an example of this already in the code base that I can use as a base?

Reply:
Have updated the PR with code that replaces this while loop